sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -774,6 +774,114 @@ case class Encode(value: Expression, charset: Expression)
 }
 
 /**
+ * Replace all substrings of str that match regexp with rep.
+ *
+ * NOTE: this expression is not THREAD-SAFE, we have to create a instance for each thread.
+ */
+case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expression)
+  extends Expression with ImplicitCastInputTypes with CodegenFallback {
```

Comment:
@chenghao-intel  the reason I want codegen is not to improve performance of this expression itself, but to prevent using fallback because once we fallback, all the children expressions will be called without codegen and require a lot of boxing.